json-endpoints/class.wpcom-json-api-get-site-v1-2-endpoint.php

```diff
@@ -59,6 +60,16 @@ class WPCOM_JSON_API_GET_Site_V1_2_Endpoint extends WPCOM_JSON_API_GET_Site_Endp
 	function callback( $path = '', $blog_id = 0 ) {
 		add_filter( 'sites_site_format', array( $this, 'site_format' ) );
 
+		// Apply filter here, return same error as switch_to_blog_and_validate_user if blog is not found.
```

Comment:
We won't be able to merge this to Jetpack as is, as `require_lib` or `Site_Filter` do not exist in Jetpack. I would recommend adding an `IS_WPCOM` check or a check for the function and the class before you use them.